front/src/applications/operationalStudies/hooks/useScenarioData.ts

```diff
@@ -0,0 +1,207 @@
+import { useSelector } from 'react-redux';
+
+import { osrdEditoastApi, type TrainScheduleResult } from 'common/api/osrdEditoastApi';
+import useInfraStatus from 'modules/pathfinding/hooks/useInfraStatus';
+import { setFailure } from 'reducers/main';
+import {
+  updateSelectedTrainId,
+  updateTrainIdUsedForProjection,
+} from 'reducers/osrdsimulation/actions';
+import { getSelectedTrainId, getTrainIdUsedForProjection } from 'reducers/osrdsimulation/selectors';
+import { useAppDispatch } from 'store';
+import { castErrorToFailure } from 'utils/error';
+
+import useLazyLoadTrains from './useLazyLoadTrains';
+import useScenario from './useScenario';
+import useSimulationResults from './useSimulationResults';
+import selectTrainAndPathForProjection from '../helpers/selectTrainAndPathForProjection';
+
+const useScenarioData = () => {
+  const dispatch = useAppDispatch();
+  const trainIdUsedForProjectionInStore = useSelector(getTrainIdUsedForProjection);
+  const selectedTrainIdInStore = useSelector(getSelectedTrainId);
+
+  const [trainSchedules, setTrainSchedules] = useState<TrainScheduleResult[]>();
+  const [trainIdUsedForProjection, setTrainIdUsedForProjection] = useState<number>();
+  const [selectedTrainId, setSelectedTrainId] = useState<number>();
+  const [trainIdsToFetch, setTrainIdsToFetch] = useState<number[]>();
+  const [trainIdsToRemove, setTrainIdsToRemove] = useState<number[]>([]);
+
+  const scenario = useScenario();
+
+  const { infra, isInfraLoaded, reloadCount } = useInfraStatus();
+
+  const { data: timetable } = osrdEditoastApi.endpoints.getV2TimetableById.useQuery(
+    { id: scenario?.timetable_id as number },
+    {
+      skip: !scenario,
+    }
+  );
+
+  const { data: rawTrainSchedules, error: fetchTrainSchedulesError } =
+    osrdEditoastApi.endpoints.postV2TrainSchedule.useQuery(
+      {
+        body: {
+          ids: timetable?.train_ids as number[],
+        },
+      },
+      {
+        skip: !timetable || !timetable.train_ids.length,
+      }
+    );
+
+  const { data: conflicts } = osrdEditoastApi.endpoints.getV2TimetableByIdConflicts.useQuery(
+    { id: scenario?.timetable_id as number, infraId: scenario?.infra_id as number },
+    { skip: !scenario }
+  );
+
+  const { data: projectionPath } = osrdEditoastApi.endpoints.getV2TrainScheduleByIdPath.useQuery(
+    {
+      id: trainIdUsedForProjection as number,
+      infraId: scenario?.infra_id as number,
+    },
+    {
+      skip: !scenario || !trainIdUsedForProjection,
+    }
+  );
+
+  const simulationResults = useSimulationResults();
+
+  const {
+    trainScheduleSummaries,
+    projectedTrains,
+    setTrainScheduleSummariesById,
+    setProjectedTrainsById,
+  } = useLazyLoadTrains({
+    infraId: scenario?.infra_id,
+    trainIdsToFetch,
+    setTrainIdsToFetch,
+    path: (projectionPath?.status === 'success' && projectionPath) || undefined,
+    trainSchedules,
+  });
+
+  useEffect(() => {
+    if (!timetable?.train_ids || !rawTrainSchedules) {
```

Comment:
Nit: I think `!timetable?.train_ids` is unnecessary here: the logic inside doesn't need it, and `rawTrainSchedules` can't be set when the timetable is missing.

Reply:
Done, thanks :)